crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +137,207 @@ impl Persistence {
             .await
             .map_err(Error::DbError)
     }
+
+    /// Get auction data.
+    pub async fn get_auction(
+        &self,
+        auction_id: domain::auction::Id,
+    ) -> Result<domain::settlement::Auction, error::Auction> {
+        let mut ex = self.postgres.pool.begin().await.context("begin")?;
+
+        let deadline = {
+            let scores = database::settlement_scores::fetch(&mut ex, auction_id)
+            .await
+            .context("fetch scores")?
+            // if score is missing, no competition / auction exist for this auction_id
+            .ok_or(error::Auction::MissingDeadline)?;
```

Comment:
Is `MissingDeadline` the right error for this? Isn't it more a missing solution or missing auction?

I think this method should return errors in ascending severity
1. There is simply no auction with this id
2. Something else about recovering the auction went wrong
3. Auction exists but there is no solution for this auction